rust/module.rs

```diff
@@ -746,18 +837,19 @@ pub fn module_misc_device(ts: TokenStream) -> TokenStream {
             kernel::prelude::module! {{
                 type: {module},
                 name: b\"{name}\",
-                author: b\"{author}\",
-                description: b\"{description}\",
+                {author}
+                {description}
                 license: b\"{license}\",
-                params: {{}},
+                {alias}
             }}
         ",
         module = module,
-        type_ = type_,
-        name = name,
-        author = author,
-        description = description,
-        license = license
+        type_ = info.type_,
+        name = info.name,
+        author = info.author.map(|v| format!("author: b\"{}\",", v)).unwrap_or_else(|| "".to_string()),
+        description = info.description.map(|v| format!("description: b\"{}\",", v)).unwrap_or_else(|| "".to_string()),
+        alias = info.alias.map(|v| format!("alias: b\"{}\",", v)).unwrap_or_else(|| "".to_string()),
```

Comment:
Perhaps we could factor this out, in particular when we get new specialized `module_*!` macros, but it is not a big deal.